app/controllers/FurloughPeriodQuestionController.scala

```diff
@@ -46,49 +48,51 @@ class FurloughPeriodQuestionController @Inject()(
 
   val form = formProvider()
 
-  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    getRequiredAnswers(FurloughStartDatePage, FurloughStatusPage) { (furloughStart, furloughStatus) =>
-      val preparedForm = request.userAnswers.get(FurloughPeriodQuestionPage) match {
-        case None        => form
-        case Some(value) => form.fill(value)
-      }
+  def onPageLoad(): Action[AnyContent] = (identify andThen feature(FastTrackJourneyFlag) andThen getData andThen requireData).async {
+    implicit request =>
+      getRequiredAnswers(FurloughStartDatePage, FurloughStatusPage) { (furloughStart, furloughStatus) =>
+        val preparedForm = request.userAnswers.get(FurloughPeriodQuestionPage) match {
+          case None        => form
+          case Some(value) => form.fill(value)
+        }
 
-      furloughStatus match {
-        case FurloughOngoing => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, None)))
-        case FurloughEnded =>
-          getAnswer(FurloughEndDatePage) match {
-            case Some(furloughEnd) => Future.successful(Ok(view(preparedForm, furloughStart, furloughStatus, Some(furloughEnd))))
-            case None =>
-              Logger.error("expecting FurloughEndDate in mongo when its furlough ended, but not found")
-              Future.successful(Redirect(routes.ErrorController.somethingWentWrong()))
-          }
+        furloughStatus match {
```

Comment:
related to above comment - here you have a function with nested pattern matching - don't you think is doing too much and part of it could be extracted?